pages/api/get-polymarket-link.ts

```diff
@@ -0,0 +1,51 @@
+import { hexString } from "helpers/validators";
+import { NextApiRequest, NextApiResponse } from "next";
+import { type, create } from "superstruct";
+import { HttpError } from "./_common";
+import { buildSearchParams } from "helpers/util/buildSearchParams";
+
+const querySchema = type({
+  questionId: hexString(),
+});
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export default async function handler(
+  request: NextApiRequest,
+  response: NextApiResponse
+) {
+  try {
+    const query = create(request.query, querySchema);
+    const questionId = query?.questionId;
+
+    if (!questionId) {
+      throw new HttpError({
+        status: 400,
+        message: "Invalid Param: questionId must be valid hex string",
+      });
+    }
+
+    const data = (await fetch(
+      `https://gamma-api.polymarket.com/markets?${buildSearchParams({
+        question_ids: questionId,
+      })}`
+    ).then((res) => res.json())) as Array<{
+      events: Array<{
+        slug: string;
+      }>;
+    }>;
+
+    const slug = data?.[0]?.events?.[0]?.slug;
+
+    console.log(`Found polymarket slug for question ID: ${questionId}`, slug);
+
+    response.setHeader("Cache-Control", "max-age=0, s-maxage=2592000");
+    response.status(200).send({ slug });
+  } catch (e) {
+    console.error(e);
```

Comment:
might want to just add a label to this error like 'Error fetching data for poly question"

Reply:
https://github.com/UMAprotocol/voter-dapp-v2/pull/356/commits/587a1a7bd68c66679cb38458b5f80e76e5b40c0f